dashboard/test/ui/test_status.js

```diff
@@ -289,20 +299,41 @@ function refresh() {
   let lastRefreshEpochSeconds = Math.floor(lastRefreshTime.getTime()/1000);
   let newTime = new Date();
   fetch(`${API_BASEPATH}/${S3_PREFIX}/since/${lastRefreshEpochSeconds}`, { mode: 'no-cors' })
-    .then(response => response.json())
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error(`While fetching updates, "${response.url}" returned ${response.status}.`);
+    })
     .then(json => {
-      json.forEach(object => {
-        let test = testFromS3Key(object.key);
-        if (test) {
-          test.setLastModified(new Date(object.last_modified));
-        }
-      });
-      lastRefreshTime = newTime;
-      lastRefreshTimeLabel.textContent = 'Updated ' + lastRefreshTime.toTimeString();
```

Comment:
Using `Promise.all` to wait on all status updates before we decide whether we've had a successful update or a partial update.